packages/flutter/test/animation/animation_controller_test.dart

```diff
@@ -37,4 +37,28 @@ void main() {
     expect(didComplete, isTrue);
     expect(didDismiss, isTrue);
   });
+
+  test("Receives status callbacks for forward and reverse", () {
+    WidgetFlutterBinding.ensureInitialized();
+    AnimationController controller = new AnimationController(
+      duration: const Duration(milliseconds: 100)
+    );
+    List<AnimationStatus> log = <AnimationStatus>[];
+    controller.addStatusListener((AnimationStatus status) {
+      log.add(status);
+    });
+
+    expect(log, equals([]));
+    controller.forward();
+    expect(log, equals([AnimationStatus.forward]));
+    controller.reverse();
+    expect(log, equals([AnimationStatus.forward, AnimationStatus.dismissed]));
+    controller.reverse();
+    expect(log, equals([AnimationStatus.forward, AnimationStatus.dismissed]));
+    log = <AnimationStatus>[];
+    controller.forward();
+    expect(log, equals([AnimationStatus.forward]));
+    controller.forward();
+    expect(log, equals([AnimationStatus.forward]));
```

Comment:
maybe also pump a couple of times and make sure that doesn't cause any further status listeners to fire